src/coreclr/nativeaot/System.Private.CoreLib/src/System/Diagnostics/StackTrace.CoreRT.cs

```diff
@@ -35,9 +34,34 @@ private void InitializeForCurrentThread(int skipFrames, bool needFileInfo)
             IntPtr[] stackTrace = new IntPtr[frameCount];
             int trueFrameCount = RuntimeImports.RhGetCurrentThreadStackTrace(stackTrace);
             Debug.Assert(trueFrameCount == frameCount);
+            skipFrames += CalculateFramesToSkip(stackTrace, trueFrameCount);
```

Comment:
Instead of this guess work, it would be better to have control about the exact number of frames to skip, adding NoInlining as necessary, similar to how we do it in `Environment.StackTrace` or how Mono does it: https://github.com/dotnet/runtime/blob/01b7e73cd378145264a7cb7a09365b41ed42b240/src/mono/System.Private.CoreLib/src/System/Diagnostics/StackFrame.Mono.cs#L31 . I think it would be fine to switch regular CoreCLR to the same plan too.

Reply:
I wanted to do that at first but that would also mean adding NoInlining to the Constructors.

```
Debug win
   at System.Runtime.RuntimeExports.RhGetCurrentThreadStackTrace(IntPtr[]) + 0xa0
   at System.Diagnostics.StackTrace.InitializeForCurrentThread(Int32, Boolean) + 0x7b
   at System.Diagnostics.StackTrace..ctor(Int32, Boolean) + 0x95
   at BringUpTest.FilterWithStackTrace(Exception) + 0x3b
```
```
Checked win
at System.Diagnostics.StackTrace.InitializeForCurrentThread(Int32, Boolean) + 0x5c
   at BringUpTest.FilterWithStackTrace(Exception) + 0x28
```

Reply:
Yes, it is fine to add `NoInlining` to the constructors.

Reply:
(and refactor things as necessary to have known number of stackframes in all cases)

Reply:
I would need to update the constructors in Runtime [libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs](https://github.com/dotnet/runtime/blob/798949a6f8013fc457cecca1f291a930670cd913/src/libraries/System.Private.CoreLib/src/System/Diagnostics/StackTrace.cs#L33), correct?

Reply:
Yep. It is fine to do the changes like this in this runtimelab branch. We will integrate them up to dotnet/runtime later.

Reply:
updated, still have one problem the Pinvoke gets inlined, if you can point me to the right location in the ILCompiler i would take a look (would be nice to always inline them since it confuses the visual studio debugger), otherwise i changed the guess work to only test for that one extra frame and remove it.

Reply:
I think you should be make the PInvoke to be always non-inlined by marking the method with NoInlining (or maybe also NoOptimization).

Alternatively, you can pass in address of something in the current stack frame into RhpCalculateStackTrace and skip all frames with stack pointer less than that address.

Reply:
NoInlining and NoOptimization had no effect, i dont think its considered either since the code is generated in place of the normal call.
Will pass in an address.